org/collections/Employee.java

```diff
@@ -0,0 +1,226 @@
+        System.out.println("Ex8. Sum of all Salaries: " + sumOfAllEmployees);
+        System.out.println("-------------------");
+        //(ex.9)
+        Map<String, List<String>> salariesPerDepartment = returnAllSalaryPerDepartment(employeeList);
+        System.out.println("Ex9. Display all salaries per Department:");
+        salariesPerDepartment.forEach((department, salaries) ->
+        System.out.println("Department: " + department + "," + salaries));
+
+    }
+    //(ex.1)
+    public static int getMaxSalaryForDepartment(ArrayList<Employee> employees, String department) {
+        int maxSalary = 0;
+        for (Employee i : employees) {
+            if (i.getDepartment() == department) {
+                if (i.getSalary() > maxSalary) {
+                    maxSalary = i.getSalary();
+                }
+            }
+        }
+        return maxSalary;
+    }
+    //(ex.2)
+    public static double getAverageSalaryForDepartment(ArrayList<Employee> employees, String department) {
+        int totalSalary = 0;
+        int count = 0;
+
+        for (Employee i : employees) {
+            if (i.getDepartment() == department) {
+                count ++;
+                if (i.getSalary() > totalSalary) {
+                    totalSalary = totalSalary + i.getSalary();
+                }
+            }
+        }
+        return totalSalary / count;
+    }
+    //(ex.3)
+    public static int getNumberOfEmployees(ArrayList<Employee> employees, String department) {
+        int count = 0;
+        for (Employee i : employees) {
+            if(i.getDepartment() == department) {
+                count ++;
+            }
+        }
+        return count;
+    }
+    //(ex.4)
+    public static int getNumberOfEmployeeExceptIT(ArrayList<Employee> employees, String exceptDepartment) {
+        int totalEmployee = 0;
+        String department;
+        for (Employee i : employees) {
+            department = i.getDepartment();
+            if(!"IT".equalsIgnoreCase(department) && !exceptDepartment.equalsIgnoreCase(department)) {
+                totalEmployee++;
+            }
+        }
+        return totalEmployee;
+    }
+    //(ex.5)
+    public static Map<String, List<Employee>> groupByJobDepartment(List<Employee> employeeList) {
+        return employeeList.stream()
+                .collect(Collectors.groupingBy(Employee::getDepartment));
+    }
+    //(ex.6)
+    public static void printEmployeesJoinedAfter2020WithSalaryAbove1000(List<Employee> employeeList) {
+        Date dateAfter = new Date(120, 0, 1);
+        for (Employee employee : employeeList) {
+            if (employee.getJoinDate().after(dateAfter) && employee.getSalary() > 1000) {
+                System.out.println(employee);
+            }
+        }
+    }
+    //(ex.7)
+    public static Map<String, Integer> getLast3Employees(List<Employee> employees) {
+        List<Employee> sortedEmployees = employees.stream()
```

Comment:
great Job here 

Now we can work on optimizing the code, to avoid using additional variables. Please take a look at the below code:
Map<String, Integer> employeeMap = employees.stream()
                .sorted((e1, e2) -> e2.getJoinDate().compareTo(e1.getJoinDate()))
                .limit(3)
                .collect(Collectors.toMap(Employee::getName, employee -> employee.getAge()));

        return employeeMap;